sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -46,79 +47,107 @@
         this.channelMono = Objects.requireNonNull(responseChannelMono, "'responseChannelMono' cannot be null.");
         this.scheduler = Objects.requireNonNull(scheduler, "'scheduler' cannot be null.");
+        this.cbsBasedTokenManager = Objects.requireNonNull(cbsBasedTokenManager,
+            "'cbsBasedTokenManager' cannot be null.");
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public Mono<ServiceBusReceivedMessage> peek() {
+    public Mono<ServiceBusReceivedMessage> peek(long fromSequenceNumber) {
 
-        int maxMessages = 2;
-        long fromSequenceNumber = 1;
-        final Map<String, Object> properties = new HashMap<>();
-        properties.put(MANAGEMENT_ENTITY_TYPE_KEY, MANAGEMENT_EVENTHUB_ENTITY_TYPE);
-        properties.put(MANAGEMENT_ENTITY_NAME_KEY, topicOrQueueName);
-        properties.put(MANAGEMENT_OPERATION_KEY, READ_OPERATION_VALUE);
-
-        return peek(properties, ServiceBusReceivedMessage.class, maxMessages, fromSequenceNumber, null)
+        return peek(new HashMap<>(), ServiceBusReceivedMessage.class, 1, fromSequenceNumber, null)
+            .last()
             .publishOn(scheduler);
     }
 
-    private <T> Mono<T> peek(Map<String, Object> appProperties, Class<T> responseType,
-                             int maxMessages, long fromSequenceNumber, UUID sessionId) {
-
-        final String tokenAudience = tokenManagerProvider.getScopesFromResource(topicOrQueueName);
-
-        return tokenProvider.getToken(new TokenRequestContext().addScopes(tokenAudience)).flatMap(accessToken -> {
-            appProperties.put(MANAGEMENT_SECURITY_TOKEN_KEY, accessToken.getToken());
-
-            // set mandatory application properties for AMQP message.
-            appProperties.put(MANAGEMENT_OPERATION_KEY, PEEK_OPERATION_VALUE);
-            appProperties.put(MANAGEMENT_SERVER_TIMEOUT, "" + 1000 * 30);
-
-            final Message request = Proton.message();
-            final ApplicationProperties applicationProperties = new ApplicationProperties(appProperties);
-            request.setApplicationProperties(applicationProperties);
-
-            // set mandatory properties on AMQP message body
-            HashMap<String, Object> requestBodyMap = new HashMap<>();
-            requestBodyMap.put(ServiceBusConstants.REQUEST_RESPONSE_FROM_SEQUENCE_NUMBER, fromSequenceNumber);
-            requestBodyMap.put(ServiceBusConstants.REQUEST_RESPONSE_MESSAGE_COUNT, maxMessages);
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<ServiceBusReceivedMessage> peek() {
 
-            if (!Objects.isNull(sessionId)) {
-                requestBodyMap.put(ServiceBusConstants.REQUEST_RESPONSE_SESSION_ID, sessionId);
-            }
+        return peek(this.lastPeekedSequenceNumber.get() + 1);
+    }
 
-            request.setBody(new AmqpValue(requestBodyMap));
+    private <T> Flux<T> peek(Map<String, Object> appProperties, Class<T> responseType,
```

Comment:
AppProperties as a parameter isn't necessary, you always pass the same parameter.